docs/en/pact-reference.md

```diff
@@ -19,265 +19,792 @@ blockchain applications with just the `pact` tool.
 
-```javascript
-{
-  "nonce": "[nonce value, needs to be unique for every call]",
-  "payload": {
-    "exec": {
-      "code": "[pact code to be executed]",
-      "data": {
-        /* arbitrary user data to accompany code */
-      }
+Transaction nonce values. Needs to be unique for every call.
+
+###### `"meta"`
+_type:_ **object** `required`
+
+Platform-specific metadata.
+
+###### `"signers"` {#cmd-signers}
+_type:_ **[ object ]** `required`
+
+List of JSON object with information on the signers authenticating the
+current payload. This signer object has the following attributes:
+
+```yaml
+name: "scheme"                # Signer's cryptographic signature scheme.
+type: enum (string)           # "ED25519" or "ETH" (Ethereum's ECDSA scheme).
+required: false               # Defaults to "ED25519".
+```
+```yaml
+name: "pubKey"                # Public key of signing keypair.
+type: string (base16)         # Must be valid public key for specified `scheme`.
+required: true
+```
+```yaml
+name: "addr"                  # Address derived from public key.
+type: string (base16)         # Must be valid address for specified `scheme`.
+required: false               # Defaults to ED25519's address representation,
+                              # which is the full public key.
+```
+
+###### `"payload"`
+_type:_ [exec](#exec-payload) **or** [cont](#cont-payload) payload `required`
+
+The `cmd` field supports two types of JSON payloads:
+the [exec payload](#exec-payload) and the [cont payload](#cont-payload).
+
+##### Example `cmd`
+```JSON
+// encoded with `exec` payload
+"{
+  \"payload\":{\"exec\":{\"data\":null,\"code\":\"(+ 1 2)\"}},
+  \"signers\":[{
+    \"addr\":\"368820f80c324bbc7c2b0610688a7da43e39f91d118732671cd9c7500ff43cca\",
+    \"scheme\":\"ED25519\",
+    \"pubKey\":\"368820f80c324bbc7c2b0610688a7da43e39f91d118732671cd9c7500ff43cca\"
+    }],
+  \"meta\":{
+    \"gasLimit\":1000,
+    \"chainId\":\"0\",
+    \"gasPrice\":1.0e-2,
+    \"sender\":\"sender00\"
+    },
+  \"nonce\":\"\\\"2019-06-20 20:56:39.509435 UTC\\\"\"
+}"
+```
+
+#### The `exec` payload {#exec-payload}
+
+The `exec` payload holds executable code and data. The [send](#send) and
+[local](#local) endpoints support this payload type in the
+`cmd` field. The format of the JSON to be encoded is as follows:
+
+##### Attributes
+
+###### `"exec"`
+_type:_ **object** `required`
+
+JSON object representing the exec payload. It has the following child attributes:
+
+```yaml
+name: "code"           # Pact code to be executed.
+type: string
+required: true
+```
+```yaml
+name: "data"          # Arbitrary user data to accompany code.
+type: object          # Must be `null` or any valid JSON.
+required: false       # This data will be injected into the scope of the
+                      # pact execution.
+```
+
+##### Example `exec` payload
+```JSON
+// encoded `exec` payload
+"{
+  \"payload\":{
+	  \"exec\":{
+		  \"data\":null,
+		  \"code\":\"(+ 1 2)\"
+	  }
+	},
+}"
```

Comment:
This looks good, but it looks like there's some escaping in inconsistent places. Any ideas?